panther_battery/test/test_roboteq_battery.cpp

```diff
@@ -0,0 +1,258 @@
+  driver_state_->rear.current = current;
+
+  battery_->Update(stamp, false);
+  battery_state_ = battery_->GetBatteryMsg();
+}
+
+void TestRoboteqBattery::TestDefaultBatteryStateMsg(
+  const uint8_t power_supply_status, const uint8_t power_supply_health)
+{
+  // const values
+  EXPECT_TRUE(std::isnan(battery_state_.temperature));
+  EXPECT_TRUE(std::isnan(battery_state_.capacity));
+  EXPECT_FLOAT_EQ(20.0, battery_state_.design_capacity);
+  EXPECT_EQ(BatteryStateMsg::POWER_SUPPLY_TECHNOLOGY_LION, battery_state_.power_supply_technology);
+  EXPECT_TRUE(panther_utils::test_utils::CheckNaNVector(battery_state_.cell_voltage));
+  EXPECT_TRUE(panther_utils::test_utils::CheckNaNVector(battery_state_.cell_temperature));
+  EXPECT_TRUE(battery_state_.present);
+  EXPECT_EQ("user_compartment", battery_state_.location);
+
+  // variable values
+  EXPECT_TRUE(std::isnan(battery_state_.voltage));
+  EXPECT_TRUE(std::isnan(battery_state_.current));
+  EXPECT_TRUE(std::isnan(battery_state_.percentage));
+  EXPECT_TRUE(std::isnan(battery_state_.charge));
+
+  EXPECT_EQ(power_supply_status, battery_state_.power_supply_status);
+  EXPECT_EQ(power_supply_health, battery_state_.power_supply_health);
+}
+
+void TestRoboteqBattery::TestBatteryStateMsg(
+  const float expected_voltage, const float expected_current, const float expected_percentage,
+  const uint8_t power_supply_status, const uint8_t power_supply_health)
+{
+  // const values
+  EXPECT_TRUE(std::isnan(battery_state_.capacity));
+  EXPECT_TRUE(std::isnan(battery_state_.temperature));
+  EXPECT_FLOAT_EQ(20.0, battery_state_.design_capacity);
+  EXPECT_TRUE(panther_utils::test_utils::CheckNaNVector(battery_state_.cell_voltage));
+  EXPECT_TRUE(panther_utils::test_utils::CheckNaNVector(battery_state_.cell_temperature));
+  EXPECT_EQ(BatteryStateMsg::POWER_SUPPLY_TECHNOLOGY_LION, battery_state_.power_supply_technology);
+  EXPECT_EQ("user_compartment", battery_state_.location);
+
+  // variable values
+  EXPECT_FLOAT_EQ(expected_voltage, battery_state_.voltage);
+  EXPECT_FLOAT_EQ(expected_current, battery_state_.current);
+  EXPECT_FLOAT_EQ(expected_percentage, battery_state_.percentage);
+  EXPECT_FLOAT_EQ(expected_percentage * 20.0f, battery_state_.charge);
+
+  EXPECT_TRUE(battery_state_.present);
+  EXPECT_EQ(power_supply_status, battery_state_.power_supply_status);
+  EXPECT_EQ(power_supply_health, battery_state_.power_supply_health);
+}
+
+TEST_F(TestRoboteqBattery, BatteryPresent) { EXPECT_TRUE(battery_->Present()); }
+
+TEST_F(TestRoboteqBattery, BatteryMsgUnknown)
+{
+  auto stamp = rclcpp::Time(0);
+  battery_->Reset(stamp);
+  battery_state_ = battery_->GetBatteryMsg();
+
+  TestDefaultBatteryStateMsg(
+    BatteryStateMsg::POWER_SUPPLY_STATUS_UNKNOWN, BatteryStateMsg::POWER_SUPPLY_HEALTH_UNKNOWN);
+}
+
+TEST_F(TestRoboteqBattery, BatteryMsgValues)
+{
+  const auto V_bat_full = 41.4;
+  const auto V_bat_min = 32.0;
+
+  const float voltage_1 = 35.0;
+  const float current_1 = 0.1;
```

Comment:
```suggestion
  const float V_bat_full = 41.4;
  const float V_bat_min = 32.0;

  const float voltage_1 = 35.0;
  const float current_1 = 0.1;
```
Why one was auto and one was float?